runtime/vm/createramclass.cpp

```diff
@@ -1439,15 +1439,28 @@ checkPackageAccess(J9VMThread *vmThread, J9Class *foundClass, UDATA classPreload
 }
 
 /**
- * Sets the current exception using the specified class name.
+ * Sets the current exception using the detailed error message plus the specified class name.
  */
 static void
-setCurrentExceptionForBadClass(J9VMThread *vmThread, J9UTF8 *badClassName, UDATA exceptionIndex)
+setCurrentExceptionForBadClass(J9VMThread *vmThread, J9UTF8 *badClassName, UDATA exceptionIndex, const char *utfMessage)
 {
-	J9MemoryManagerFunctions *gcFuncs = vmThread->javaVM->memoryManagerFunctions;
-	j9object_t detailMessage = gcFuncs->j9gc_createJavaLangString(vmThread, J9UTF8_DATA(badClassName), J9UTF8_LENGTH(badClassName), J9_STR_XLAT);
+	UDATA bufSize = J9UTF8_LENGTH(badClassName) + (UDATA)strlen(utfMessage) + 2;
+	char *errorBuf = NULL;
+	PORT_ACCESS_FROM_VMC(vmThread);
+
+	errorBuf = (char*)j9mem_allocate_memory(bufSize + 1, J9MEM_CATEGORY_CLASSES);
+	if (NULL == errorBuf) {
+		setNativeOutOfMemoryError(vmThread, 0, 0);
```

Comment:
It's better to throw the original exception (even without the message) then to change it to OOM.

Reply:
Already updated.